components/infobox/wikis/rocketleague/infobox_team_custom.lua

```diff
@@ -62,43 +59,21 @@ function CustomInjector:wrapErrorMessage(text)
 	return outText
 end
 
-function CustomInjector:parse(id, widgets)
-	if id == 'earnings' then
-		local earnings = Earnings.calculateForTeam({team = _team.pagename or _team.name})
-		Variables.varDefine('earnings', earnings)
-		if earnings == 0 then
-			earnings = nil
-		else
-			earnings = '$' .. Language:formatNum(earnings)
-		end
-		return {
-			Cell{
-				name = 'Earnings',
-				content = {
-					earnings
-				}
-			}
-		}
-	end
+function CustomInjector:parse(_, widgets)
 	return widgets
 end
 
 function CustomTeam:addToLpdb(lpdbData, args)
-	lpdbData.earnings = Variables.varDefault('earnings', 0)
 	if not String.isEmpty(args.teamcardimage) then
 		lpdbData.logo = 'File:' .. args.teamcardimage
 	elseif not String.isEmpty(args.image) then
 		lpdbData.logo = 'File:' .. args.image
 	end
 
-	lpdbData.extradata = {
-		rating = Variables.varDefault('rating'),
-		tier = string.lower(args.tier or '')
-	}
+	lpdbData.extradata.rating = Variables.varDefault('rating')
+	lpdbData.extradata.tier = string.lower(args.tier or '')
 	for year = _START_YEAR, _CURRENT_YEAR do
-		local id = args.id or _team.pagename
-		local earningsInYear = Template.safeExpand(mw.getCurrentFrame(), 'Total earnings of', {year = year, id})
-		lpdbData.extradata['earningsin' .. year] = (earningsInYear or ''):gsub(',', ''):gsub('$', '')
+		lpdbData.extradata['earningsin' .. year] = '$' .. (lpdbData.extradata['earningsin' .. year] or 0)
```

Comment:
'$' should not be added If I'm reading the old code correctly

Reply:
And the reason we need this is because some querying module in RL expects earningsinyearX to always exist?

Reply:
how lpdb data looks on RL with the old implementation:

![Screenshot 2021-11-19 16 25 09](https://user-images.githubusercontent.com/75081997/142647712-aad3358c-97b3-4d59-8a75-d65275d92d17.png)

Reply:
I'm pretty sure that's must be a bug that's slipped through. The earnings stats pages on RL have stopped working for teams.
https://liquipedia.net/rocketleague/Statistics/2021

Reply:
Old template based version definitely removed $'s.
![image](https://user-images.githubusercontent.com/3426850/142660353-81090f5c-0ceb-4178-981d-608f689ed34e.png)
And current active module (attempts) to remove the $ sign (`:gsub('$', '')`), but failing?
![image](https://user-images.githubusercontent.com/3426850/142661260-f036e644-8f56-4266-85d8-e60563038e2e.png)

Reply:
yeah looks like it
will check if the query is used somewhere else and adjust accordingly (most likely can kick the custom stuff)

Reply:
Found the reason for it failing.

> string.gsub has two return values. The first is result string, while the second is the total number of matches that occurred.

Reply:
> Found the reason for it failing.
> 
> > string.gsub has two return values. The first is result string, while the second is the total number of matches that occurred.

one of the reasons i like using `string.gsub(str, a, b)` over `str:gsub(a, b)`

checked (via insource search) all queries modules and templates that use `earningsinXXXX` and they use `or 0` so we can kick the entire custom part from RL
this fixes the bug AND cleans the code and standards